src/bus/index.spec.js

```diff
@@ -0,0 +1,265 @@
+        bus._baseSocketUrl = expectedHost;
+
+        socket = new WebSocket(
+          `${expectedHost}/organizations/${expectedOrganization.id}/stream`
+        );
+        expectedSocket = new Socket(socket, expectedOrganization.id);
+
+        bus._sockets[expectedOrganization.id] = expectedSocket;
+
+        promise = bus.connect(expectedOrganization.id);
+      });
+
+      it('fulfills the promise', function() {
+        expect(promise).to.be.fulfilled;
+      });
+
+      it('does not fetch an api token', function() {
+        return promise.then(() => {
+          expect(sdk.auth.getCurrentApiToken).to.not.be.called;
+        });
+      });
+
+      it('resolves the promise with the existing socket', function() {
+        return promise.then((socket) => {
+          expect(socket).to.deep.equal(expectedSocket);
+          expect(socket).to.be.an.instanceof(Socket);
+          expect(socket._organizationId).to.equal(expectedOrganization.id);
+          expect(socket._socket.url).to.equal(
+            `${expectedHost}/organizations/${expectedOrganization.id}/stream`
+          );
+        });
+      });
+    });
+
+    context(
+      'when a websocket does not already exist for the organization',
+      function() {
+        context('when successfully connecting to the message bus', function() {
+          let bus;
+          let expectedApiToken;
+          let promise;
+          let sdk;
+
+          beforeEach(function() {
+            expectedApiToken = faker.internet.password();
+
+            sdk = {
+              ...baseSdk,
+              auth: {
+                ...baseSdk.auth,
+                getCurrentApiToken: this.sandbox
+                  .stub()
+                  .resolves(expectedApiToken)
+              }
+            };
+
+            bus = new Bus(sdk, baseRequest);
+            bus._baseSocketUrl = expectedHost;
+
+            promise = bus.connect(expectedOrganization.id);
+          });
+
+          context('when intially opening the websocket', function() {
+            it('gets an api token', function() {
+              return promise.then(() => {
+                expect(sdk.auth.getCurrentApiToken).to.be.calledWith(
+                  'contxtAuth'
+                );
+              });
+            });
+
+            it('stores a copy of the websocket', function() {
+              return promise.then(() => {
+                const socket = bus._sockets[expectedOrganization.id];
+
+                expect(socket).to.be.an.instanceof(Socket);
+                expect(socket._organizationId).to.equal(
+                  expectedOrganization.id
+                );
+                expect(socket._socket.url).to.equal(
+                  `${expectedHost}/organizations/${
+                    expectedOrganization.id
+                  }/stream`
+                );
+              });
+            });
+          });
+
+          context('when the websocket closes', function() {
```

Comment:
This _feels_ like a weird place to put this -- is there another spot or level it could be at?